### PR TITLE
1754 script convert sample to address update

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Checklist
Reminder: If any change made to this repo affects the acceptance tests then the version pin in the Pipfile in [census-rm-acceptance-tests]() needs to be updated.
* [ ] Updated census-rm-acceptance-tests version pin (if applicable)

# Motivation and Context
We need to be able to update case data from an amended sample file by looking up the case ID's and building a validate address update file. 

# What has changed
* Add script to build address update file from sample format
* Tests
* Update readme

# How to test?
Check and run the script locally

# Links
https://trello.com/c/z7hB1yJr/1754-script-to-convert-sample-file-into-address-update-file-ie-with-case-ids-5

### CODE REVIEW

File: toolbox/tests/bulk_processing/test_build_address_update_from_sample.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    patch_requests.get.return_value.raise_for_status.side_effect = HTTPError()
+
+    patch_requests.get.return_value.status_code = 500
+
+    with pytest.raises(HTTPError):
+        generate_address_update_file(TEST_SAMPLE_FILE)
+
+
+@patch('toolbox.bulk_processing.build_address_update_from_sample.requests')
+def test_rasises_if_uprn_has_multiple_cases(patch_requests):
```
Comment (LeoKHoward) [inline toolbox/tests/bulk_processing/test_build_address_update_from_sample.py]: Spelling mistake

File: toolbox/bulk_processing/build_address_update_from_sample.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    output_address_update_file.unlink()
+                    raise
+                address_update_row = {k: row[k] for k in fields_to_copy}
+                address_update_row[case_id_fieldname] = case_id
+                output_writer.writerow(address_update_row)
+
+    return output_address_update_file
+
+
+def get_case_id_from_case_api(uprn, line_number):
```
Comment (NickGrantX) [inline toolbox/bulk_processing/build_address_update_from_sample.py:52]: I estimate this job could take over 2 hours to run (20 milliseconds per roundtrip times 414,000)... in which case a glitch with the Case API would be annoying. Should this retry? Kinda a nice to have I guess

Comment (AdamHawtin) [reply toolbox/bulk_processing/build_address_update_from_sample.py:52]: Fair point, I was also unsure about failing out on errors. On the one hand the file should have been pre-validated, on the other failing the whole thing in the 100th minute when 99.9% of the data could have been fine is frustrating.

Comment (NickGrantX) [reply toolbox/bulk_processing/build_address_update_from_sample.py:52]: up to you... plenty of our other scripts have similar glitch problems

File: toolbox/tests/bulk_processing/test_build_address_update_from_sample.py (hunk 1 above)
Comment (NickGrantX) [reply toolbox/tests/bulk_processing/test_build_address_update_from_sample.py]: CENSUS CANCELLED!

**Review by hbrace — COMMENTED** [review]
Would it make sense to keep processing even when an error is found and output any issues to a file for investigation? Otherwise from what I can see, this would need to be re run every time we find and rectify an issue.

**Comment by AdamHawtin** [comment]
@hbrace The scope on the card didn't include that and we have separate validation to run beforehand.

File: toolbox/tests/bulk_processing/test_build_address_update_from_sample.py (hunk 1 above)
Comment (LeoKHoward) [reply toolbox/tests/bulk_processing/test_build_address_update_from_sample.py]: If we cannot speak well English then how useful are we?

**Comment by LeoKHoward** [comment]
I am happy to approve after manually testing but I understand why Hugh and Nick made their comments

File: toolbox/tests/bulk_processing/test_build_address_update_from_sample.py (hunk 1 above)
Comment (NickGrantX) [reply toolbox/tests/bulk_processing/test_build_address_update_from_sample.py]: curious... you didn't comment on all the greengrocers' apostrophes in the README

**Review by NickGrantX — APPROVED** [review]
I see no reason to block this. Good work 👍

**Review by LeoKHoward — APPROVED** [review]

File: toolbox/tests/bulk_processing/test_build_address_update_from_sample.py (hunk 1 above)
Comment (AdamHawtin) [reply toolbox/tests/bulk_processing/test_build_address_update_from_sample.py]: Well now I have to fix those as well 😂

**Review by hbrace — APPROVED** [review]

**Pull request merged**